app/scripts/views/mixins/password-mixin.js

```diff
@@ -12,6 +12,7 @@ define(function (require, exports, module) {
 
   module.exports = {
     events: {
+      'focus input.password': 'onPasswordFocus',
```

Comment:
I have not yet tested it, but it seems like you are not properly handling the case where the user types a password on the signin/signup page, goes to the TOS/PP agreement, and then comes back. Or, types a password into signin, goes to signup. The show button should be visible.